helix-contract/contracts/utils/BoundedVec.sol

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.10;
+
+library BoundedVec {
+    struct Vec {
+        bytes32[] _values;
+        uint256 _capacity;
+        uint256 _start;
+        uint256 _length;
+    }
+
+    function _setCapacity(Vec storage vec, uint256 _capacity) internal {
+        require(vec._capacity == 0, "BoundedVec:can't resize vec");
+        vec._capacity = _capacity;
+    }
+
+    function _length(Vec storage vec) internal view returns(uint256) {
+        return vec._length;
+    }
+
+    function _start(Vec storage vec) internal view returns(uint256) {
+        return vec._start;
+    }
+
+    function _at(Vec storage vec, uint256 index) internal view returns(bytes32) {
+        require(index < vec._length, "BoundedVec:index overflow");
+        uint256 pos = index + vec._start;
+        if (pos >= vec._capacity) {
+            return vec._values[pos - vec._capacity];
+        } else {
+            return vec._values[pos];
+        }
+    }
+
+    function _push(Vec storage vec, bytes32 _value) internal {
+        uint256 size = vec._length;
+        require(size < vec._capacity, "BoundedVec:index overflow");
+        uint256 pos = vec._start + size;
+        if (pos >= vec._capacity) {
+            vec._values[pos - vec._capacity] = _value;
+        } else {
+            if (vec._values.length < vec._capacity) {
```

Comment:
Bad case:
```sh
begin:
_values: [0, 0, 0, 0]
_capacity: 4
_start: 1
_length: 0

then _capacity change to 10

if (vec._values.length < vec._capacity) is true
and vec._values.push(_value) will set _values[5] = _value

in this case: _values[1] = _value
```

Reply:
we are not allowed to change the _capacity when it's initialized because the requirement https://github.com/helix-bridge/contracts/blob/xiaoch05-parachain-bridge/helix-contract/contracts/utils/BoundedVec.sol#L13

Reply:
That's OK.